### PR TITLE
Update autocompletion for ECMAScript2021

### LINKED ISSUE #1358: Autocomplete incorrect with ECMAScript 2021+
## The problem

MainUI rule editor is suggesting incorrect autocomplete entries with `application/javascript;version=ECMAScript-2021`:

image

## Expected behavior

MainUI rule editor would not suggest incorrect autocomplete entries.

For example, with ECMAScript 2021+ (aka `application/javascript;version=ECMAScript-2021`), there should be instead `event.oldItemState`, not `event.previousState`.

## Steps to reproduce

Create rule, select item trigger, and use action with type "ECMAScript 262 Edition 11". 

## Your environment

```yaml
runtimeInfo:
  version: 3.2.0
  buildString: Release Build
locale: fi-FI
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 11.0.11
  javaVendor: AdoptOpenJDK
  javaVendorVersion: AdoptOpenJDK-11.0.11+9
  osName: Linux
  osVersion: 5.10.63-v7l+
  osArchitecture: arm
  availableProcessors: 4
  freeMemory: 125179096
  totalMemory: 959447040
bindings:
  - fmiweather
  - modbus
  - systeminfo
  - zwave
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: false
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    pixelRatio: 1.25
    prefersColorScheme: light
  isSecureContext: false
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 4
    language: en-GB
    languages:
      - en-GB
      - en
      - en-US
      - fi
    onLine: true
    platform: Linux x86_64
  screen:
    width: 3072
    height: 1728
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like
    Gecko) Chrome/100.0.4896.88 Safari/537.36
timestamp: 2022-04-24T11:07:19.945Z

```

## Browser console

```txt

```

### PR DESCRIPTION
**Pull request by florian-h05**
@ghys 

Fixes #1358 

This PR updates the openhab-js tern definitions for the breaking changes and improvements of the latest openhab-js release (2.0.0) as well as some other improvements.

The `event` properties are fixed.

The `time` and the `things` namespaces and the cloud notification actions are also added.

I have tested the change, everything works but I was unable to use the ItemConfig as parameter type for `addItem` and `replaceItem`.

Signed-off-by: Florian Hotze <florianh_dev@icloud.com>

### CODE REVIEW

**Comment by ghys** [comment]
Thanks!

> I have tested the change, everything works but I was unable to use the ItemConfig as parameter type for addItem and replaceItem.

I tried with the obvious:
```json
            "addItem": {
                "!doc": "Creates a new Item within openHAB (see JS Scripting docs/Standard Library/Items).",
                "!type": "fn(itemConfig: ItemConfig) -> Item",
                "!url": "https://openhab.org/addons/automation/jsscripting/#itemconfig"
            },
```
and it seems to work?

image

(however you shouldn't denote optional keys with '?' because they will be taken literally and added when you autocomplete).

File: bundles/org.openhab.ui/web/src/assets/openhab-js-tern-defs.json [hunk 1]
```diff
@@ -201,6 +293,15 @@
                 "!type": "fn(key: string) -> ?"
             }
         },
+        "TimeNamespace": {
+            "toZDT": {
+                "!doc": "Converts nearly any representation of a time to a time.ZonedDateTime (see JS Scripting Docs/Standard Library/Time)",
+                "!type": "fn(time: any) -> time.ZonedDateTime"
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/assets/openhab-js-tern-defs.json:299]: Please note that `any` is a TypeScript keyword, here is should be `?`.

**Comment by florian-h05** [comment]
> I tried with the obvious:
> …
> and it seems to work?

Interesting, I was trying this yesterday evening and it did not work. I will recheck ASAP.

> (however you shouldn't denote optional keys with '?' because they will be taken literally and added when you autocomplete).

Okay, I‘ll fix that.

File: bundles/org.openhab.ui/web/src/assets/openhab-js-tern-defs.json (hunk 1 above)
Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/assets/openhab-js-tern-defs.json:299]: Okay thanks.
I will do this ASAP.

**Review by ghys — APPROVED** [review]
@florian-h05 I addressed the points above in my last commit so that this PR is merged in time for 3.3.RC1. Thanks!

**Comment by florian-h05** [comment]
@ghys Thank you very much!

I was planning to address your comments this evening when I am back home, as I was not at home this week and did not have access to my machine.

**Pull request merged**